contracts/base/Dispatcher.sol

```diff
@@ -269,9 +271,40 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, V4SwapRout
                     }
 
                     (success, output) = address(V3_POSITION_MANAGER).call(inputs);
+                } else if (command == Commands.V4_INITIALIZE_POOL) {
+                    bytes4 selector;
+                    assembly {
+                        selector := calldataload(inputs.offset)
+                    }
+                    if (selector != PoolInitializer.initializePool.selector) {
+                        revert InvalidAction(selector);
+                    }
+                    (success, output) = address(V4_POSITION_MANAGER).call(inputs);
                 } else if (command == Commands.V4_POSITION_CALL) {
                     // should only call modifyLiquidities() to mint
-                    // do not permit or approve this contract over a v4 position or someone could use this command to decrease, burn, or transfer your position
+                    bytes4 selector;
+                    assembly {
+                        selector := calldataload(inputs.offset)
+                    }
+                    if (selector != V4_POSITION_MANAGER.modifyLiquidities.selector) {
+                        revert InvalidAction(selector);
+                    }
+
+                    bytes calldata slice = inputs[4:];
+                    (bytes calldata actions,) = slice.toBytes(0).decodeActionsRouterParams();
```

Comment:
oh you never use the parameters its decoding... i feel like it would be possible instead (and more efficient) to try this...
`slice.toBytes(0).toBytes(0)`